app/src/main/java/com/dimelo/sampleapp/MyApplication.java

```diff
@@ -0,0 +1,28 @@
+package com.dimelo.sampleapp;
+
+import android.app.Application;
+import android.util.Log;
+
+import com.google.android.gms.maps.MapsInitializer;
+import com.google.android.gms.maps.OnMapsSdkInitializedCallback;
+
+public class MyApplication extends Application implements OnMapsSdkInitializedCallback {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        MapsInitializer.initialize(getApplicationContext(), MapsInitializer.Renderer.LATEST, this);
+    }
+
+    @Override
+    public void onMapsSdkInitialized(MapsInitializer.Renderer renderer) {
+        switch (renderer) {
+            case LATEST:
+                Log.d("MapsDemo", "The latest version of the renderer is used.");
+                break;
+            case LEGACY:
```

Comment:
yes exactly, just we test

Reply:
yeah this `case` will only be reached in case we're on a device that does not support the new renderer (android version too old, not enough memory, etc...)